File: c#/WebApplication1/Recipe.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;

namespace WebApplication1
{
    public class Recipe
    {
        public string NameRecipe;
        public Category CategoryRecipe;
        public int Time;
        public int Difficulty;
        public DateTime DateAdd;
        public List<string> Ingredients;
        public List<string> Preparation;
        public string UserName;
        public string Image;
        public bool View;
    }
}

File: src/app/class/recipe.ts
import { Category } from './Category';

export class recipe{
    constructor(public NameRecipe?:string,public CategoryRecipe?:Category,public Time?:number, public Difficulty?:number,public DateAdd?:Date,
        public  Ingredients?:string[],public Preparation?:string[],public UserName?:string,public Image?:string,public View?:boolean
        )
     {}
}
